### ISSUE TITLE
JavaDocs for most .java files

### ISSUE THREAD

**Issue by Jason-Whitmore** [Jason-Whitmore/jpf#7]
The standard javadoc style of documentation should be good at explaining what's happening in the classes, constructors, and methods. These should be completed for most .java files.

Checklist:
-ActivationFunction (Complete)
-Add (Complete)
-CrossEntropy (Complete)
-CSVWriter (Complete)
-Dense (Complete)
-Examples
-Input (Complete)
-Layer (Complete)
-LeakyReLU (Complete)
-LinearAlgebra
-Linear (Complete)
-LinearModel (Complete)
-Loss (Complete)
-Model (Complete)
-MSE (Complete)
-NeuralNetwork (Complete)
-Optimizer (Complete)
-PolynomialModel (Complete)
-RMSProp (Complete)
-SGD (Complete)
-Sigmoid (Complete)
-SimpleModel (Complete)
-SoftmaxLayer (Complete)
-Tanh (Complete)
-Utility (Complete)
